frontend/packages/kubevirt-plugin/src/plugin.tsx

```diff
@@ -205,6 +205,35 @@ const plugin: Plugin<ConsumedExtensions> = [
       required: FLAG_KUBEVIRT,
     },
   },
+  {
+    type: 'Page/Route',
+    properties: {
+      exact: true,
+      path: [
+        '/k8s/ns/:ns/virtualmachines',
+        '/k8s/all-namespaces/virtualmachines',
+        '/k8s/ns/:ns/virtualmachineinstances',
+        '/k8s/all-namespaces/virtualmachineinstances',
```

Comment:
IMO we shouldn't overwrite virtualmachineinstances path as it might be useful to have a view just for simple CRDs (eg you just want to drill to the yaml of the VMI once it comes up)

I don't see a big problem of having path for simple VirtualMachine CR too since we are not advertising this path anywhere (just the CRDs page), but I guess we could redirect to virtualization

can we get more opinions on this? @matthewcarleton @jelkosz

Reply:
We need to override the vanila vm and vmtemplate links because:
a - autogenerated urls use the object type to generate the link, link to vm and vmtemplate lists, if we do not redirect this links, we will send users to the wrong page.
b - the vm and vmtemplate list views do not give the user extra data.

vmi list view different, the list view of vmis show all vims while the virtualization list view show only vmis that are not owned by vms.

@matthewcarleton we need you help with the vmi list hiding issue:
a - is it ok to hide the vanila vmi list, without giving users alternative way to see a list of all vmis ?
b - can we add an option to the vm/vmi list to show only vms or only vmi's in a way that when showing only vmi's we will show **all** of them including those with vm owners ?

Reply:
> We need to override the vanila vm and vmtemplate links because:
> a - autogenerated urls use the object type to generate the link, link to vm and vmtemplate lists, if we do not redirect this links, we will send users to the wrong page.
> b - the vm and vmtemplate list views do not give the user extra data.
> 
> vmi list view different, the list view of vmis show all vims while the virtualization list view show only vmis that are not owned by vms.
> 
> @matthewcarleton we need you help with the vmi list hiding issue:
> a - is it ok to hide the vanila vmi list, without giving users alternative way to see a list of all vmis ?

> b - can we add an option to the vm/vmi list to show only vms or only vmi's in a way that when showing only vmi's we will show **all** of them including those with vm owners ?

Both of these questions sound like filtering questions. I am reluctant to say we should allow the user to filter the list based on VMI/VM because we'd have to customize the filtering mechanism right? I know that OCP will be delivering a new toolbar which we could make use of then. Do we have use cases where we know this type of capability is needed?

Reply:
> I am reluctant to say we should allow the user to filter the list based on VMI/VM because we'd have to customize the filtering mechanism right? 

it's already customized ... so no problem here :-)
we should open a BZ saying we want to filter by vm vs vmi ...

> Do we have use cases where we know this type of capability is needed?

no, just guesses based on what we ( not so avarage users :-) ) think will be nice :-)